server/workerTest.js

```diff
@@ -23,28 +34,30 @@ if (isMainThread) {
     }
 
     for (let i = 0; i < numRecords; i++) {
-      records.push({
-        title: faker.commerce.productName(),
-        brand: brands[ Math.floor( Math.random() * brands.length ) ],
-        department: departments[ Math.floor( Math.random() * departments.length ) ],
-        price: Number(faker.commerce.price(0, 100)) - Math.ceil(Math.random() * 5) / 100,
-        imageUrl: `https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/target-images/${i % 50}.jpg`,
-        productUrl: `/${i % 100 + 1}`
-      });
+      records +=
+        `${faker.commerce.productName()}, ${brands[ Math.floor( Math.random() * brands.length ) ]}, ${departments[ Math.floor( Math.random() * departments.length ) ]}, ${Number(faker.commerce.price(0, 100)) - Math.ceil(Math.random() * 5) / 100}, https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/target-images/${i % 50}.jpg, /${i % 100 + 1}\n`
     }
-    return records;
+
+    return records
 
   };
   let t1 = performance.now()
-  const records = generateRecords(250);
-  const recordsJSON = JSON.stringify(records);
-  fs.writeFile(`../data/random-data-${id}.json`, recordsJSON, (err, results) => {
+  let records = generateRecords(1250000);
+  fs.appendFile(`../data/random-data.csv`, records, (err, results) => {
     if (err) {
       console.log(err);
     } else {
-      let t2 = performance.now()
-      const timeTaken = t2 - t1;
-      console.log(`worker ${id} took ${timeTaken} milliseconds to write ${records.length} records`)
+      records = generateRecords(1250000);
```

Comment:
clarifying question: So each worker is creating 2.5M records but you've split this work in half.  Why did you choose to do it this way?

Reply:
@kcmcminn Hey Kristina! My computer didn't have enough RAM to store 2.5m * 4 workers records at once, so it would kill my node process. I split them into two, because then my RAM would never deal with more than 5m in RAM at once. That said, I actually just optimized this by creating an appending write stream that eliminates the need to hold any object (array) in memory! I'm going to push it up now if you wouldn't mind reviewing - the code looks cleaner too :)